Rakefile

```diff
@@ -116,14 +131,19 @@ task 'dlls' do
   end
 end
 
-desc 'Build the native gem file under rake_compiler_dock'
-task 'gem:native' do
+desc 'Build the native gem file under rake_compiler_dock. Optionally one can pass argument to build only native gem for a chosen platform.'
+task 'gem:native', [:plat] do |t, args|
   verbose = ENV['V'] || '0'
 
   grpc_config = ENV['GRPC_CONFIG'] || 'opt'
   ruby_cc_versions = ['3.0.0', '2.7.0', '2.6.0', '2.5.0'].join(':')
+  selected_plat = "#{args[:plat]}"
 
   if RUBY_PLATFORM =~ /darwin/
+    if !selected_plat.empty? && selected_plat != 'darwin'
+      fail "Cannot pass platform as an argument when on Darwin."
+    end
+
     FileUtils.touch 'grpc_c.32.ruby'
```

Comment:
nit: since we need to touch these files in most cases, can we move these touch commands up to lines above `if RUBY_PLATFORM =~ /darwin/` ?

Then, I think we can delete the `FilesUtils.touch` lines everywhere else.

Reply:
I think moving FilesUtils.touch to the top of the native_gem task can potentially mask problems.
If I create grpc_c.*.ruby using touch at the beginning of the native_gem rule and there's some misconfiguration of the "dlls" task, I can potentially end up with the windows gems containing the empty placeholder files instead of the real dlls and it could happen silently.
So I prefer not doing touching the dll files that will potentially need to be buildt by the "dlls" task.
PTAL at the approach I took to simplify the Rakefile, but hopefully without masking potential problems.